tinytroupe/agent/grounding.py

```
@@ -0,0 +1,272 @@
+        self.add_web_urls(self.web_urls)
+    
+    def add_web_urls(self, web_urls:list) -> None:
+        """ 
+        Adds the data retrieved from the specified URLs to grounding.
+        """
+        filtered_web_urls = [url for url in web_urls if url not in self.loaded_web_urls]
+        for url in filtered_web_urls:
+            self._mark_web_url_as_loaded(url)
+
+        if len(filtered_web_urls) > 0:
+            new_documents = SimpleWebPageReader(html_to_text=True).load_data(filtered_web_urls)
```

Ruff (0.8.2)

Undefined name `SimpleWebPageReader`

(F821)